src/containers/App/reducer.ts

```diff
@@ -0,0 +1,136 @@
+import { VARIANT_NOTICE, TYPE_LOCAL } from 'containers/Notification/constants';
+import { APPLY_FILTER } from 'signals/incident-management/constants';
+import type { Action } from 'types';
+import type { AppActionTypes } from './actions';
+import {
+  LOGIN_FAILED,
+  LOGOUT_FAILED,
+  LOGOUT,
+  AUTHORIZE_USER,
+  SHOW_GLOBAL_NOTIFICATION,
+  RESET_GLOBAL_NOTIFICATION,
+  UPLOAD_PROGRESS,
+  UPLOAD_SUCCESS,
+  UPLOAD_FAILURE,
+  SET_SEARCH_QUERY,
+  RESET_SEARCH_QUERY,
+  GET_SOURCES,
+  GET_SOURCES_FAILED,
+  GET_SOURCES_SUCCESS,
+} from './constants';
+import type { AppState } from './types';
+
+export type ApplyFilterActionType = Action<typeof APPLY_FILTER, never>;
+
+// The initial state of the App
+export const initialState: AppState = {
+  loading: false,
+  error: false,
+  upload: {},
+  user: {
+    permissions: [],
+    roles: [],
+  },
+  notification: {
+    message: '',
+    title: '',
+    variant: VARIANT_NOTICE,
+    type: TYPE_LOCAL,
+  },
+  searchQuery: '',
+  sources: [],
+};
+
+type ReducerActionTypes = AppActionTypes | ApplyFilterActionType | Action<null, never>;
```

Comment:
What's the use for `Action<null, never>` ?

Reply:
This is for handling the default branch of the switch. Maybe we can find a better pattern for this. Any suggestions are welcome!